library/x509_crl.c

```diff
@@ -418,12 +442,37 @@ int mbedtls_x509_crl_parse_der( mbedtls_x509_crl *chain,
      *                                   -- if present, MUST be v2
      *                        } OPTIONAL
      */
-    if( ( ret = x509_get_entries( &p, end, &crl->entry ) ) != 0 )
+    entry = mbedtls_calloc( 1, sizeof( mbedtls_x509_crl_entry ) );
+    if( entry == NULL )
+    {
+        mbedtls_x509_crl_free( crl );
+        return( MBEDTLS_ERR_X509_ALLOC_FAILED );
+    }
+    count = 1;
+
+    if( ( ret = x509_get_entries( &p, end, entry, &count ) ) != 0 )
```

Comment:
nice one! thanks! although the mistake I made is pretty stupid :) I'll rearrange this code a bit, also considering the previous and the next comment.